Support border/padding/margin on RenderMathMLOperator

fred-wang opened the pull request:
#### 9a01858119398922a3baa934a4eb225786d15cda
<pre>
Support border/padding/margin on RenderMathMLOperator
<a href="https://bugs.webkit.org/show_bug.cgi?id=276357">https://bugs.webkit.org/show_bug.cgi?id=276357</a>

Reviewed by Rob Buis.

This patch adds support for border/padding/margin on `&lt;mo&gt;` elements
for which useMathOperator() returns true and therefore uses use
special layout and painting instead of deferring to RenderMathMLToken.

* LayoutTests/TestExpectations: Mark borders/paddings/margins reftests
  for largeop/stretchy operators as passing.
* Source/WebCore/rendering/mathml/RenderMathMLOperator.cpp:
(WebCore::RenderMathMLOperator::stretchTo): Include borders/paddings in
  logical width/height.
(WebCore::RenderMathMLOperator::computePreferredLogicalWidths): Include
  borders/paddings in preferred width. Children are text nodes, no need
  to consider margins.
(WebCore::RenderMathMLOperator::layoutBlock): Include borders/paddings
  for in logical width/height. Also call recomputeLogicalWidth() to set
  the inline margins on the element (block margins are handled by the
  parent). Children are text nodes, no need to consider margins for them.
(WebCore::RenderMathMLOperator::firstLineBaseline const):
  Include border/padding in the ascent.
(WebCore::RenderMathMLOperator::paint): Shift painting by the left/top
  border/padding.

Canonical link: <a href="https://commits.webkit.org/280833@main">https://commits.webkit.org/280833@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b99c7408ae378bbbfa0a9c9967f96710d9220d4f

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/57765 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/37093 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/10241 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/61387 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/8210 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/59893 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/44729 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/8398 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/46802 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/5825 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/59795 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/34784 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/49913 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/27627 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/31554 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/7206 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/7214 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/53506 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/7476 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/63070 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/1679 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/7553 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/54021 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/1685 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/49924 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/54139 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/12779 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/1419 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/32922 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/34008 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/35092 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/33753 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rwlbuis reviewed (approved):
LGTM. In commit message "so use" -> "therefore uses"

fred-wang committed:
Support border/padding/margin on RenderMathMLOperator
https://bugs.webkit.org/show_bug.cgi?id=276357

Reviewed by Rob Buis.

This patch adds support for border/padding/margin on `<mo>` elements
for which useMathOperator() returns true and therefore uses use
special layout and painting instead of deferring to RenderMathMLToken.

* LayoutTests/TestExpectations: Mark borders/paddings/margins reftests
  for largeop/stretchy operators as passing.
* Source/WebCore/rendering/mathml/RenderMathMLOperator.cpp:
(WebCore::RenderMathMLOperator::stretchTo): Include borders/paddings in
  logical width/height.
(WebCore::RenderMathMLOperator::computePreferredLogicalWidths): Include
  borders/paddings in preferred width. Children are text nodes, no need
  to consider margins.
(WebCore::RenderMathMLOperator::layoutBlock): Include borders/paddings
  for in logical width/height. Also call recomputeLogicalWidth() to set
  the inline margins on the element (block margins are handled by the
  parent). Children are text nodes, no need to consider margins for them.
(WebCore::RenderMathMLOperator::firstLineBaseline const):
  Include border/padding in the ascent.
(WebCore::RenderMathMLOperator::paint): Shift painting by the left/top
  border/padding.

Canonical link: https://commits.webkit.org/280833@main
